.github/workflows/ci.yml

```diff
@@ -0,0 +1,146 @@
+name: CI
+
+on: [pull_request, push]
+
+jobs:
+  Build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    container:
+      image: elementary/docker:unstable
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Install Depedencies
+        env:
+          dependency_packages: libvala-dev valac meson
+        run: apt-get update && apt-get -y install ${dependency_packages}
+
+      - name: Build
+        run: meson build && ninja -C build
+
+      - name: Upload
+        uses: actions/upload-artifact@v1
+        with:
+          name: vala-lint
+          path: ./
+
+  Test:
+    name: Test Build
+    runs-on: ubuntu-latest
```

Comment:
I think `runs-on` is not needed if you specify `container`

Reply:
Nope. Still need `runs-on`.